cpp/tests/table/table_tests.cpp

```diff
@@ -167,3 +168,39 @@ TEST_F(TableTest, ConcatenateTables)
 
   cudf::test::expect_tables_equal(*concat_table, gold_table);
 }
+
+TEST_F(TableTest, ConcatenateTablesWithOffsets)
```

Comment:
Better to add one more test case for null case, string column with offset has nulls elements.